src/test/java/org/folio/rest/impl/AbstractInstanceRecordsAPITest.java

```diff
@@ -0,0 +1,123 @@
+package org.folio.rest.impl;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.sqlclient.Tuple;
+
+import static org.folio.rest.api.StorageTestSuite.getVertx;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Supplier;
+import javax.ws.rs.core.Response;
+import org.folio.rest.RestVerticle;
+import org.folio.rest.api.StorageTestSuite;
+import org.folio.rest.api.TestBase;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.AdditionalAnswers;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+@RunWith(VertxUnitRunner.class)
+public class AbstractInstanceRecordsAPITest extends TestBase {
+  private static Map<String,String> okapiHeaders = Collections.singletonMap(
+      RestVerticle.OKAPI_HEADER_TENANT, StorageTestSuite.TENANT_ID);
+
+  public static void fetchRecordsByQuery(String sql, RoutingContext routingContext,
+      Supplier<Tuple> paramsSupplier, Handler<AsyncResult<Response>> asyncResultHandler) {
+
+    new AbstractInstanceRecordsAPI(){}.fetchRecordsByQuery(
+        sql, paramsSupplier, routingContext, okapiHeaders,
+        asyncResultHandler, Vertx.vertx().getOrCreateContext(), null);
+  }
+
+  @Test
+  public void test500(TestContext testContext) {
```

Comment:
I find it helpful to understand the scenario the test is trying to describe from the name.

I think this test demonstrates that a 500 response should be sent when an error occurs during fetching of records.

```suggestion
  public void shouldRespondWith500StatusWhenErrorsOccursWhilstFetchingRecords(TestContext testContext) {
```

Reply:
The test method names have been renamed.